### PR TITLE
Client server app

### PR DESCRIPTION
**Pull request by Ionel14**

### CODE REVIEW

File: client/smarthomeconnect.cpp [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+	serverAddress.sin_port = htons(8080); 
+	serverAddress.sin_addr.s_addr = INADDR_ANY; 
+
+	if(connect(*clientSocket, (struct sockaddr*)&serverAddress, sizeof(serverAddress)) == -1)
+    {
+        std::cerr << "Failed to bind socket" << std::endl;
+        return;
+    } 
+
+    while (true)
```
Comment (enachebogdan) [inline client/smarthomeconnect.cpp:29]: This is no big deal, but personally I'd move this logic outside the function(maybe into a run function or something similar). SOLID - S, your function should only do one thing: start a connection.(you don't need to change it, just keep in mind)

File: client/smarthomeconnect.cpp [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+        std::cerr << "Failed to receive data from server" << std::endl;
+        return;
+    }
+
+    std::cout << "\nReceived from server:\n" << buffer << std::endl;
+}
+
+std::string SmartHomeConnect::RequestString()
+{
+    std::cout<<"Menu:\n";
```
Comment (enachebogdan) [inline client/smarthomeconnect.cpp:67]: Again, nothing needs to be changed here, just think about this: how would you decouple the following things from your functions/class - displaying the menu(UI handling), input handling and your actual class logic.

File: client/smarthomeconnect.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+class SmartHomeConnect
+{
+public:
+    void StartConnection();
+
+
+private:
+    std::string RequestString();
+    std::string StatusAction();
```
Comment (enachebogdan) [inline client/smarthomeconnect.h:22]: function name should be a verb/action

File: server/room.h [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+public:
+    Room(const std::string &name = "");
+
+    void AddSensor(MyUniquePtr<Sensor> &&sensor);
+    bool DeleteSensor(const std::string &name);
+
+    void AddDevice(MyUniquePtr<Device> &&device);
+    bool DeleteDevice(const std::string &name);
+
+    std::string& GetName();
```
Comment (enachebogdan) [inline server/room.h:24]: when you return references to class members, make sure that you return them as const.

File: server/room.cpp [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+void Room::AddSensor(MyUniquePtr<Sensor> &&sensor)
+{
+    sensors_.emplace_back(std::move(sensor));
+
+}
+
+bool Room::DeleteSensor(const std::string &name)
+{
+    for (int i = 0; i < sensors_.size(); i++)
```
Comment (enachebogdan) [inline server/room.cpp:16]: use a better way to iterate.

File: server/room.h [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    void SetName(const std::string &name);
+
+    const std::vector<StatusObject*> GetSensors();
```
Comment (enachebogdan) [inline server/room.h:27]: here you are not actually returning a const reference of your vector.

**Review by enachebogdan — APPROVED** [review]
Your homework looks good, nice job. I only had some minor comments, nothing important.

**Comment by Ionel14** [comment]
> Your homework looks good, nice job. I only had some minor comments, nothing important.

Thanks, I looked at the comments and I will try to pay more attention to those aspects in the future

**Pull request merged**